README.adoc

```diff
@@ -0,0 +1,471 @@
+.  Marker file to temporarily disable the feature for a specific Maven
+build. +
+Typically used to disable a reporter for a specific build that would
+generate too much data for the default configuration of the reporter
+(e.g. too many generated artifacts...) or to workaround a bug in the
+"[.code]``withMaven``" waiting for a fix. These marker file must be located in
+the home directory of the build.
+
+==== Implicit or Explicit activation of Publishers
+
+By default, all the publishers are enabled by default.
+
+It is possible to change the default activation of a publisher
+navigating to the "Global Tool Configuration" screen.
+
+It is possible to disable the default activation of publishers on a
+specific "``+withMaven(){...}+``" step using the
+"[.code]``publisherStrategy='EXPLICIT'``" attribute in the step
+"``+withMaven(publisherStrategy='EXPLICIT'){...}+``". The publishers can
+then be enabled explicitly in the "``+withMaven(){...}+``" step using the
+"publishers" attribute
+
+=== Default Configuration
+
+Default Maven settings can be defined globally and at the folder level.
+
+==== Global Default Configuration
+
+In the "Global Tool Configuration" screen
+
+* Maven settings and Maven global settings
+* Publishers settings: enable/disable publishers...
+
+image::docs/images/pipeline-maven-plugin-global-tools-configuration.png[]
+
+==== Folder Level Configuration
+
+In the Folder "configuration" screen
+
+* Maven settings and Maven global settings
+
+image:docs/images/pipeline-maven-folder-level-configuration.png[]
+
+=== Trigger downstream pipeline when a snapshot is built (since 3.0.0)
+
+Trigger downstream pipeline that depend on Maven artifact generated by
+upstream pipelines.
+
+image:docs/images/information.svg[(info)] Notes
+
+* The upstream artifact must be generated in a "[.code]``withMaven(){}``"
+wrapping step to be detected by the triggering system
+* The downstream pipeline must have selected the build trigger "Build
+whenever a SNAPSHOT dependency is built"
+ ** The build trigger can be defined at the pipeline level ("Build
+Triggers"), at the multibranch pipeline level ("Scan Repository
+Triggers") or at the GitHub Organization / Bitbucket Project
+level ("Scan Organizations Triggers")
+* You have to manually trigger once the upstream pipeline and the
+downstream pipeline so that the link between the pipelines based on
+the SNAPSHOT dependency is established
+* The dependency graph is, for the moment, exclusively stored in an H2
+embedded database
+("[.code]``$JENKINS_HOME/jenkins-jobs/jenkins-jobs.mv.db``"). Support for an
+external H2 database and then for alternate databases (PostgreSQL)
+is on the roadmap
+(see https://github.com/jenkinsci/pipeline-maven-plugin/blob/pipeline-maven-3.0.0-beta-1/jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/dao/PipelineMavenPluginH2Dao.java[PipelineMavenPluginH2Dao.java])
```

Comment:
```suggestion
[NOTE]
====
* The upstream artifact must be generated in a "[.code]``withMaven(){}``" wrapping step to be detected by the triggering system
* The downstream pipeline must have selected the build trigger "Build whenever a SNAPSHOT dependency is built"
 ** The build trigger can be defined at the pipeline level ("Build Triggers"), at the multibranch pipeline level ("Scan Repository Triggers") or at the GitHub Organization / Bitbucket Project level ("Scan Organizations Triggers")
* You have to manually trigger once the upstream pipeline and the downstream pipeline so that the link between the pipelines based on the SNAPSHOT dependency is established
* The dependency graph is, for the moment, exclusively stored in an H2 embedded database ("[.code]``$JENKINS_HOME/jenkins-jobs/jenkins-jobs.mv.db``"). Support for an external H2 database and then for alternate databases (PostgreSQL) is on the roadmap (see https://github.com/jenkinsci/pipeline-maven-plugin/blob/pipeline-maven-3.0.0-beta-1/jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/dao/PipelineMavenPluginH2Dao.java[PipelineMavenPluginH2Dao.java])
====
```